adminme doesnt work

Hi !
As said on [this page](https://github.com/Half-Shot/matrix-appservice-discord/blob/master/docs/howto.md) we can become matrix admin with adminme. I tried unsuccessfully.

This is the exact command I typed :

```
[ T0MuX > ~/matrix-appservice-discord ]
$ npm run adminme -- -r '!dNutYBqCnHmvrjUvlU:mux.re' -u '@t0:mux.re' -p '100' -c '/home/t0mux/matrix-appservice-discord/config.yaml'                                             

> matrix-appservice-discord@0.2.0 adminme /home/t0mux/matrix-appservice-discord
> node ./build/tools/adminme.js "-r" "!dNutYBqCnHmvrjUvlU:mux.re" "-u" "@t0:mux.re" "-p" "100" "-c" "/home/t0mux/matrix-appservice-discord/config.yaml"                          

SyncApi.sync: starting with sync token null
Got push rules
/sync error M_UNKNOWN: Internal server error
{ [M_UNKNOWN: Internal server error]
  errcode: 'M_UNKNOWN',
  name: 'M_UNKNOWN',
  message: 'Internal server error',
  data: { errcode: 'M_UNKNOWN', error: 'Internal server error' },
  httpStatus: 500 }
Starting keep-alive
/home/t0mux/matrix-appservice-discord/node_modules/olm/olm.js:19
"undefined"!==typeof module&&(module.exports=a);process.on("uncaughtException",function(a){if(!(a instanceof T))throw a;});a.inspect=function(){return"[Emscripten Module object]"}}else if(ma)a.print||(a.print=print),"undefined"!=typeof printErr&&(a.printErr=printErr),a.read="undefined"!=typeof read?read:function(){throw"no read() available (jsc?)";},a.readBinary=function(a){if("function"===typeof readbuffer)return new Uint8Array(readbuffer(a));a=read(a,"binary");z("object"===typeof a);return a},"undefined"!=                    
                                                                                                                ^                                                                

TypeError: Cannot set property 'access_token' of undefined
    at MatrixHttpApi._request (/home/t0mux/matrix-appservice-discord/node_modules/matrix-js-sdk/lib/http-api.js:620:34)                                                          
    at MatrixHttpApi.requestOtherUrl (/home/t0mux/matrix-appservice-discord/node_modules/matrix-js-sdk/lib/http-api.js:558:21)                                                   
    at MatrixHttpApi.request (/home/t0mux/matrix-appservice-discord/node_modules/matrix-js-sdk/lib/http-api.js:443:21)                                                           
    at SyncApi._pokeKeepAlive (/home/t0mux/matrix-appservice-discord/node_modules/matrix-js-sdk/lib/sync.js:892:23)                                                              
    at ontimeout (timers.js:427:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
    at Timer.processTimers (timers.js:212:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! matrix-appservice-discord@0.2.0 adminme: `node ./build/tools/adminme.js "-r" "!dNutYBqCnHmvrjUvlU:mux.re" "-u" "@t0:mux.re" "-p" "100" "-c" "/home/t0mux/matrix-appservice-discord/config.yaml"`
npm ERR! Exit status 7
npm ERR!
npm ERR! Failed at the matrix-appservice-discord@0.2.0 adminme script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.                                                                               

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/t0mux/.npm/_logs/2018-07-18T11_58_14_083Z-debug.log
```

And this is the content of the mentioned log : 
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'run',                                                                                                                                                           
1 verbose cli   'adminme',
1 verbose cli   '--',
1 verbose cli   '-r',
1 verbose cli   '!dNutYBqCnHmvrjUvlU:mux.re',                                                                                                                                    
1 verbose cli   '-u',
1 verbose cli   '@t0:mux.re',
1 verbose cli   '-p',              
1 verbose cli   '100',                                           
1 verbose cli   '-c',
1 verbose cli   '/home/t0mux/matrix-appservice-discord/config.yaml' ]
2 info using npm@6.2.0                                          
3 info using node@v10.6.0                                                                                                                                                         
4 verbose run-script [ 'preadminme', 'adminme', 'postadminme' ]                                                                                                                   
5 info lifecycle matrix-appservice-discord@0.2.0~preadminme: matrix-appservice-discord@0.2.0                                                                                     
6 info lifecycle matrix-appservice-discord@0.2.0~adminme: matrix-appservice-discord@0.2.0                                                                                        
7 verbose lifecycle matrix-appservice-discord@0.2.0~adminme: unsafe-perm in lifecycle true                                                                                       
8 verbose lifecycle matrix-appservice-discord@0.2.0~adminme: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/t0mux/matrix-appservice-discord/node_mo
dules/.bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl         
9 verbose lifecycle matrix-appservice-discord@0.2.0~adminme: CWD: /home/t0mux/matrix-appservice-discord                                                                          
10 silly lifecycle matrix-appservice-discord@0.2.0~adminme: Args: [ '-c',                                                                                                        
10 silly lifecycle   'node ./build/tools/adminme.js "-r" "!dNutYBqCnHmvrjUvlU:mux.re" "-u" "@t0:mux.re" "-p" "100" "-c" "/home/t0mux/matrix-appservice-discord/config.yaml"' ]   
11 silly lifecycle matrix-appservice-discord@0.2.0~adminme: Returned: code: 7  signal: null                                                                                      
12 info lifecycle matrix-appservice-discord@0.2.0~adminme: Failed to exec adminme script
13 verbose stack Error: matrix-appservice-discord@0.2.0 adminme: `node ./build/tools/adminme.js "-r" "!dNutYBqCnHmvrjUvlU:mux.re" "-u" "@t0:mux.re" "-p" "100" "-c" "/home/t0mux/m
atrix-appservice-discord/config.yaml"`       
13 verbose stack Exit status 7
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:304:16)                                                          
13 verbose stack     at EventEmitter.emit (events.js:182:13)                                                                                                                      
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)                                                       
13 verbose stack     at ChildProcess.emit (events.js:182:13)
13 verbose stack     at maybeClose (internal/child_process.js:961:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)                                                                                    
14 verbose pkgid matrix-appservice-discord@0.2.0                                                                                                                                 
15 verbose cwd /home/t0mux/matrix-appservice-discord
16 verbose Linux 4.14.54-1-lts                      
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "adminme" "--" "-r" "!dNutYBqCnHmvrjUvlU:mux.re" "-u" "@t0:mux.re" "-p" "100" "-c" "/home/t0mux/matrix-appservice-discord/con
fig.yaml"                              
18 verbose node v10.6.0                                        
19 verbose npm  v6.2.0             
20 error code ELIFECYCLE        
21 error errno 7               
22 error matrix-appservice-discord@0.2.0 adminme: `node ./build/tools/adminme.js "-r" "!dNutYBqCnHmvrjUvlU:mux.re" "-u" "@t0:mux.re" "-p" "100" "-c" "/home/t0mux/matrix-appservic
e-discord/config.yaml"`   
22 error Exit status 7
23 error Failed at the matrix-appservice-discord@0.2.0 adminme script.
23 error This is probably not a problem with npm. There is likely additional logging output above.                                                                               
24 verbose exit [ 7, true ]
```

I don't know what is happening. Apparently, the token is missing. However, my bot and bridge are currently working fine with the good token in the configuration file.

Kind Regards

--- a/tools/adminme.ts
+++ b/tools/adminme.ts
@@ -100,9 +100,8 @@ const {appservice} = ToolsHelper.getToolDependencies(options.config, options.reg
 async function run() {
     try {
         const powerLevels = (await appservice.botIntent.underlyingClient.getRoomStateEvent(
-            options.roomId, "m.room.power_levels", "",
-        )).content;
-
+            options.roomid, "m.room.power_levels", "",
+        ));
         powerLevels.users[options.userid] = options.power;
 
         await appservice.botIntent.underlyingClient.sendStateEvent(

--- a/tools/toolshelper.ts
+++ b/tools/toolshelper.ts
@@ -12,7 +12,8 @@ export class ToolsHelper {
         config: DiscordBridgeConfig,
     } {
         const registration = yaml.safeLoad(fs.readFileSync(regFile, "utf8"));
-        const config: DiscordBridgeConfig = yaml.safeLoad(fs.readFileSync(configFile, "utf8")) as DiscordBridgeConfig;
+        const config: DiscordBridgeConfig = Object.assign(
+            new DiscordBridgeConfig(), yaml.safeLoad(fs.readFileSync(configFile, "utf8")));
         config.applyEnvironmentOverrides(process.env);
         if (registration === null) {
             throw Error("Failed to parse registration file");

